lib/config/validation.ts

```diff
@@ -147,9 +147,13 @@ export async function validateConfig(
         message: `The "${key}" object can only be configured at the top level of a config but was found inside "${parentPath}"`,
       });
     }
-    if (!isGlobalConfig) {
+
+    if (isGlobalConfig) {
+      validateGlobalConfig(key, val, optionTypes[key], warnings, currentPath);
+      continue;
```

Comment:
this skips validation for non-global options inside global config file. they shouldn't skipped